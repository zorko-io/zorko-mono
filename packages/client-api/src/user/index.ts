export * from './user.api.client';
export * from './user.axios.api.client';
