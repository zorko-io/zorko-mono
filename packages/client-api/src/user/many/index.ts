export * from './user.many.api.client'
export * from './user.axios.many.api.client'
