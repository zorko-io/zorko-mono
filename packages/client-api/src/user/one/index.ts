export * from './user.one.api.client'
export * from './user.axios.one.api.client'
