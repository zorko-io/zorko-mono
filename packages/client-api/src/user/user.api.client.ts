import { AuthApiClient } from '../auth';
import { RemoteUserApi } from '@zorko/remote-api';

export interface UserApiClient extends AuthApiClient, RemoteUserApi {}
