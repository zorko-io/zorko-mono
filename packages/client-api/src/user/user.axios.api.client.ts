import { AxiosResponse } from 'axios';
import { AuthAxiosApiClient } from '../auth';
import { UserDto } from '@zorko/dto';
import { UserApiClient } from './user.api.client';
import {
  CreateUserCollectionParams,
  CreateUserParams, DeleteUserCollectionParams, DeleteUserParams,
  ReadUserCollectionParams,
  ReadUserParams,
  UpdateUserCollectionParams, UpdateUserParams,
} from '@zorko/remote-api';

export class UserAxiosApiClient extends AuthAxiosApiClient implements UserApiClient  {

  async findMany(readParams?: ReadUserCollectionParams): Promise<UpdateUserCollectionParams> {
    const response: AxiosResponse<UpdateUserCollectionParams> = await this.http.get('/users');
    return response.data
  }

  async createOne(user: CreateUserParams): Promise<string> {
    const response: AxiosResponse<string> = await this.http.post('/users', user);
    return response.data;
  }

  async findOne(readParams: ReadUserParams): Promise<UpdateUserParams> {
    const { id }  = readParams;
    const response: AxiosResponse<UserDto> = await this.http.get(`/users/${id}`);
    return response.data;
  }

  async createMany(createParams: CreateUserCollectionParams): Promise<string> {
    return '123'
  }

  async removeMany(deleteParams: DeleteUserCollectionParams): Promise<string> {
    const response: AxiosResponse<string> = await this.http.delete('/users');
    return response.data
  }

  async removeOne(deleteParams: DeleteUserParams): Promise<void> {
    const { id } = deleteParams;
    await this.http.delete(`/users/${id}`);
  }

  updateMany(updateParams: UpdateUserCollectionParams): Promise<ReadUserCollectionParams> {
    return undefined;
  }

  updateOne(updateParams: UpdateUserParams): Promise<ReadUserParams> {
    return undefined;
  }
}

