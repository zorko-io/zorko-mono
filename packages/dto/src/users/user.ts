import { RolesEnum } from '../roles';

export interface User {
  id?: any;
  email: string;
  password?: string;

  roles?: RolesEnum[];
}
