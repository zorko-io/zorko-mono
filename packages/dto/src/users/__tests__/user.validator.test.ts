import { UserValidator } from '../user.validator';
import { User } from '../user';

describe('UserValidator', () => {
  let validator;
  let data: User;
  let actual;

  beforeEach(() => {
    validator = new UserValidator();
  });

  it('passes minimum validation', () => {
    data = { email: 'test@ffkfkf.com'};
    actual = validator.validate({
      email: 'test@ffkfkf.com'
    });

    expect(actual).toEqual(data);
  });

  it('passes max validation', () => {
    data = {
      id: '312123',
      email: 'test@ffkfkf.com',
      roles: [],
      password: '234342423434'
    };
    actual = validator.validate(data);
    expect(actual).toEqual(data);
  });

  it('fails on email', () => {
    actual = validator.validate({
      id: '312123',
      email: 'test!ffkfkf_94949494',
      roles: [],
      password: '234342423434'
    });

    expect(actual).toBeFalsy();
    expect(validator.getErrors()).toMatchSnapshot();
  });

});
