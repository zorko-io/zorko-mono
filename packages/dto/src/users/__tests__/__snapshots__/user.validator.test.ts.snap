// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserValidator fails on email 1`] = `
Object {
  "email": "WRONG_EMAIL",
}
`;
